src/core/qgsapplication.h

```diff
@@ -475,7 +475,7 @@ class CORE_EXPORT QgsApplication : public QApplication
      *
      * \since QGIS 3.26
      */
-    static int systemMemorySizeMb();
+    static unsigned long systemMemorySizeMb();
 
```

Comment:
This is fixing the problem that "int", assuming int is at least 32 bits (qgis won't build on a PDP-11 last I checked :-), can only represent 2^31 * 2*20 = 2^51 bytes of RAM?   Which is 2T?  That seems fair, as such RAM is now merely expensive and hard to find a motherboard for.    I would lean to using uint64_t, but that's aesthetics.   I might add a comment that a 32-bit signed variable could only represent memory < 2T.